coders/heic.c

```diff
@@ -299,8 +305,58 @@ static Image *ReadHEICImage(const ImageInfo *image_info,
   heif_image_release(heif_image);
   heif_image_handle_release(image_handle);
   heif_context_free(heif_context);
+
+  /*
+    There is a discrepancy between EXIF data and the actual orientation of
+    image pixels. ReadImage processes "exif:Orientation" expecting pixels to be
+    oriented accordingly. However, in HEIF the pixels are NOT rotated.
+   */
+  option=GetImageOption(image_info,"heic:preserve-orientation");
+  if (IsStringTrue(option) == MagickTrue)
+    image = CompensateOrientation(image, exception);
+  else
+    SetImageProperty(image, "exif:Orientation", "1", exception);
+
   return(GetFirstImageInList(image));
 }
+
+/* An inverse of AutoOrientImage */
+static Image *CompensateOrientation(Image *image, ExceptionInfo *exception)
```

Comment:
`Image *image` should be `const Image *image`.

Reply:
Makes sense, I will change it to a const pointer type

Reply:
Good point. I missed that, probably better to not change the signature to a const.